sources/api/prairiedog/src/bootconfig.rs

```diff
@@ -0,0 +1,410 @@
+            None
+        } else {
+            Some(kernel_params)
+        },
+        init_parameters: if init_params.is_empty() {
+            None
+        } else {
+            Some(init_params)
+        },
+    })
+}
+
+/// Given a boot config string, deserialize it to `model::BootSettings` and then serialize it back
+/// out as a JSON string for sundog consumption
+fn boot_config_to_boot_settings_json(bootconfig_str: &str) -> Result<String> {
+    // We'll only send the setting if the existing boot config file fits our settings model
+    let boot_settings = parse_boot_config_to_boot_settings(bootconfig_str)?;
+    // sundog expects JSON-serialized output
+    serde_json::to_string(&boot_settings).context(error::OutputJsonSnafu)
+}
+
+#[cfg(test)]
+mod boot_settings_tests {
+    use crate::bootconfig::{
+        boot_config_to_boot_settings_json, serialize_boot_settings_to_boot_config,
+        DEFAULT_BOOTCONFIG_STR,
+    };
+    use maplit::hashmap;
+    use model::BootSettings;
+    use serde_json::json;
+    use serde_json::value::Value;
+    use std::convert::TryInto;
+
+    #[test]
+    fn boot_settings_to_string() {
+        let boot_settings = BootSettings {
+            kernel_parameters: Some(
+                hashmap! {
+                    "console" => vec!["ttyS1,115200n8", "tty0"],
+                }
+                .into_iter()
+                .map(|(k, v)| {
+                    (
+                        k.try_into().unwrap(),
+                        v.into_iter().map(|s| s.try_into().unwrap()).collect(),
+                    )
+                })
+                .collect(),
+            ),
+            init_parameters: Some(
+                hashmap! {
+                    "systemd.log_level" => vec!["debug"],
+                    "splash" => vec![""],
+                }
+                .into_iter()
+                .map(|(k, v)| {
+                    (
+                        k.try_into().unwrap(),
+                        v.into_iter().map(|s| s.try_into().unwrap()).collect(),
+                    )
+                })
+                .collect(),
+            ),
+        };
+        let output = serialize_boot_settings_to_boot_config(&boot_settings).unwrap();
+        // Sort the entries alphabetically to keep results consistent
+        let mut lines = output
+            .lines()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>();
+        lines.sort();
+        let output = lines.join("\n");
+        assert_eq!(
+            output,
+            r#"init.splash = ""
+init.systemd.log_level = "debug"
+kernel.console = "ttyS1,115200n8", "tty0""#
+        );
+    }
+
+    #[test]
+    fn none_boot_settings_to_string() {
+        let boot_settings = BootSettings {
+            kernel_parameters: None,
+            init_parameters: None,
+        };
+        assert_eq!(
+            serialize_boot_settings_to_boot_config(&boot_settings).unwrap(),
+            r#""#
```

Comment:
Just double-checking - the kernel is fine with an empty bootconfig file? Or do we need to emit our placeholder value to avoid a warning?

The `bootconfig` tool in the kernel tree says "Error: Config data is empty" if given an empty file.

Reply:
`prairiedog` handles the case where BootSettings or if both `kernel` and `init` are empty.
It stubs in the default bootconfig file if the settings aren't there:
```
bash-5.1# apiclient get boot
{}
bash-5.1# prairiedog --log-level trace generate-boot-config
22:34:02 [TRACE] (1) schnauzer: [api/schnauzer/src/lib.rs:81] GETing from /settings?prefix=boot
...
22:34:02 [TRACE] (1) schnauzer: [api/schnauzer/src/lib.rs:95] JSON response: {}
22:34:02 [TRACE] (1) prairiedog::bootconfig: [api/prairiedog/src/bootconfig.rs:67] Serializing boot config string: 
    kernel = ""
    init = ""
...
```